testing/testservice/instrumentation.go

```diff
@@ -0,0 +1,48 @@
+package testservice
+
+import (
+	"context"
+	"io"
+)
+
+type RunConfiguration struct {
+	serviceID string
+	ctx       context.Context
+	args      []string
+	output    io.Writer
+}
+
+func (cfg *RunConfiguration) GetID() string {
+	return cfg.serviceID
+}
+
+func (cfg *RunConfiguration) GetArgs() []string {
+	return cfg.args
+}
+
+func (cfg *RunConfiguration) GetContext() context.Context {
+	return cfg.ctx
+}
+
+func (cfg *RunConfiguration) AppendArgs(args ...string) {
+	cfg.args = append(cfg.args, args...)
+}
+
+func (cfg *RunConfiguration) WrapContext(fn func(context.Context) context.Context) {
+	cfg.ctx = fn(cfg.ctx)
+}
+
+type Instrumentation interface {
```

Comment:
Wouldn't mind a TemporalInstrumentation implementation to add the following flags from the golibs temporal package:

* TemporalAddressFlag
* TemporalNamespaceFlag
* TemporalInitSearchAttributesFlag